### PR TITLE
docs: Add sample instructions on multitenancy

### LINKED ISSUE #126: [1] Test and document workload multi-tenancy
To accelerate workload multi-tenancy installation, without creating technical debt, we suggested the following solution:

* "ck8s init" multiple pairs of SC-WC clusters, let's call them P_0, P_1, P_2, ... .
* We configure the same baseDomain and opsDomain for all P_i.
* For P0 we only do "apply sc".
* We point the DNS entry for *.$baseDomain and *.$opsDomain (I still don't get the difference between the two) to one of P_0's SC worker nodes. (I operate under the assumption that XYZ does not have a decent load-balancer.)
* For all P_i with i > 0, we only do "apply wc".

Acceptance criteria:
* Please test if this works and create a Markdown document that shows how to do it. Someone familiar with Compliant Kubernetes should be able to easily follow the prose and copy-paste the code.

### PR DESCRIPTION
**Pull request by OlleLarsson**
**What this PR does / why we need it**:
To get some guidline on how to achive one serivce cluster with multiple workload clusters.

**Which issue this PR fixes**:
fixes #126 

**Special notes for reviewer**:
Don't really know what expected so i guess this is a first draft :)

**Tests**

- Metrics from both workload clusters show up in prometheus/influxDB/grafana
grafana-mc

- Logs from both workload clusters show up in elasticsearch/kibana
logs-cluster-label

**Checklist:**

- [ ] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [ ] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.

### CODE REVIEW

**Review by Xartos — COMMENTED** [review]
I think this is great! Maybe mention something like
```
This documentation is assuming you are setting up a cluster on Exoscale.
Although the same principles apply on the other clouds
```

And another thing.
Should this file live in root? or should we have some docs folder we can put this in?

**Comment by OlleLarsson** [comment]
> I think this is great! Maybe mention something like
> 
> ```
> This documentation is assuming you are setting up a cluster on Exoscale.
> Although the same principles apply on the other clouds
> ```
> 
> And another thing.
> Should this file live in root? or should we have some docs folder we can put this in?

We can perhpas simply put it in `docs/`?

File: multitenancy.md
Comment (cristiklein) [inline multitenancy.md:25]: Can you tell how `ck8s_linux_amd64` can be found?

File: multitenancy.md
Comment (cristiklein) [inline multitenancy.md:92]: I don't understand this. :( Refresh my memory, what is the `kubeapi-metrics` Ingress and why is it neded? What are the consequences of this not working?

File: multitenancy.md
Comment (cristiklein) [inline multitenancy.md:94]: Can you use the same tense as above? "Ingress of workload clusters has not been tested, but is expected to work without issues."

**Review by cristiklein — APPROVED** [review]
This is awesome!

LGTM once the minor comments are addressed.

I would very much love to have this here, to get started with operator documentation: https://compliantkubernetes.io/operator-manual/lifecycle/

File: multitenancy.md
Comment (viktor-f) [inline multitenancy.md:7]: The link to the apps repo is pointing to `main`. In the text above it seems like you say that this should point to the specific commit that you used.

File: multitenancy.md
Comment (viktor-f) [inline multitenancy.md:73]: Did you try changing the `baseDomain` of the workload cluster to something else? If that works then we could have a wildcard DNS record pointing to each workload cluster as well.

File: multitenancy.md
Comment (viktor-f) [inline multitenancy.md:75]: You are missing the step where you apply apps

File: multitenancy.md
Comment (viktor-f) [inline multitenancy.md:48]: I don't think we have the `exo` cli as a requirement anywhere. So either add a link to where one can download it or instead use our `ck8s-dns` repo.

File: multitenancy.md
Comment (viktor-f) [inline multitenancy.md:85]: Same thing here regarding `exo`.

**Review by viktor-f — CHANGES_REQUESTED** [review]
Looks nice! Found some comments

File: multitenancy.md
Comment (OlleLarsson) [reply multitenancy.md:75]: Correct

File: multitenancy.md
Comment (OlleLarsson) [reply multitenancy.md:48]: ck8s-dns is not an open project. I'll add a link to exo cli

File: multitenancy.md
Comment (OlleLarsson) [reply multitenancy.md:25]: In ck8s-cluster when you have built the cli
```
cd ck8s-cluster && make
ls dist/
```

File: multitenancy.md
Comment (OlleLarsson) [reply multitenancy.md:7]: Yes was waiting for this to be approved and merged https://github.com/elastisys/compliantkubernetes-apps/pull/145

File: multitenancy.md
Comment (OlleLarsson) [reply multitenancy.md:73]: I'll try and check :)

File: multitenancy.md
Comment (viktor-f) [reply multitenancy.md]: Good point, forgot about that.

File: multitenancy.md
Comment (OlleLarsson) [reply multitenancy.md:113]: Got the certificates to work by pointing dns to the clusters.
The kubeapi-metrics is used to expose the `/healtz` endpoint of the kubeapi-server. This endpoint is normally checked by the blackbox exporter in the service cluster.

**Comment by OlleLarsson** [comment]
I think I've addressed all you comments.
@cristiklein should I move this pr to https://github.com/elastisys/compliantkubernetes ?

File: multitenancy.md
Comment (OlleLarsson) [reply multitenancy.md]: Pointed dns to wc loadbalancer and used unique basedomain for each cluster and it seems to work. Now only opsdomain is shared between the clusters.

**Comment by cristiklein** [comment]
@OlleLarsson Yes, please. We really lack documentation in this section https://compliantkubernetes.io/operator-manual/lifecycle/

**Comment by OlleLarsson** [comment]
> @OlleLarsson Yes, please. We really lack documentation in this section https://compliantkubernetes.io/operator-manual/lifecycle/

https://github.com/elastisys/compliantkubernetes/pull/35

**Pull request closed without merging**
